test/MyChartAPITest.js

```diff
@@ -0,0 +1,417 @@
+                                            DayKey:
+                                                "SD_10033367_10098243_2008_20210403_en-US",
+                                            DurationInMinutes: 0,
+                                            SlotProviderId: "",
+                                            CanShowAppointmentTime: true,
+                                        },
+                                    ],
+                                    AmPm: "pm",
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+            DayOfWeek: 6,
+        },
+    },
+    ProviderInfoHtml: null,
+    EarliestDate: "2021-04-01",
+    LatestDate: "2021-04-08",
+    Specialty: null,
+    VisitType: "2008",
+    ErrorCode: null,
+    VisitTypeInfo: {
+        Name: "COVID-19 VACCINE 1",
+        DisplayName: "COVID-19 Vaccine 1st",
+        ID: "2008",
+        IDType: "EXTERNAL",
+        FromDaysOffset: null,
+        ToDaysOffset: null,
+        PatientInstructions: [],
+        IsAdvanced: true,
+        GroupByDepartment: true,
+        IsTelemedicine: false,
+        IsExcludedFromWizard: true,
+    },
+};
+
+const exampleWithError = {
+    AllDays: null,
+    AllDepartments: null,
+    AllProviders: null,
+    ByDateThenProviderCollated: null,
+    ProviderInfoHtml: null,
+    EarliestDate: null,
+    LatestDate: null,
+    Specialty: null,
+    VisitType: null,
+    ErrorCode: "StartDateTooFarInFuture",
+    VisitTypeInfo: null,
+};
+
+describe("AddSiteInfo", () => {
+    it("should add site info and default data to results object", () => {
+        results = {};
+        siteKey = Object.keys(exampleWithAvailability.AllDepartments)[0];
+        siteInfo = exampleWithAvailability.AllDepartments[siteKey];
+
+        mychart.AddSiteInfo(results, { default: "default" }, siteKey, siteInfo);
+
+        // should have key for department ID
+        expect(results).to.have.property(siteKey);
+
+        // should add default values
+        expect(results[siteKey].default).to.equal("default");
+
+        // should add name and address
+        expect(results[siteKey].name).to.equal(siteInfo.Name);
+        expect(results[siteKey].street).to.equal(
+            siteInfo.Address.Street.join(" ")
+        );
+        expect(results[siteKey].city).to.equal(siteInfo.Address.City);
+        expect(results[siteKey].zip).to.equal(siteInfo.Address.PostalCode);
+    });
+});
+
+describe("UpdateResults", () => {
+    it("should add site info and default data to results object", () => {
+        results = {};
+        mychart.UpdateResults(results, exampleWithAvailability);
+
+        for (const [key, info] of Object.entries(results)) {
+            if (key == 10098243) {
+                // this is the only site that has availability
+                expect(info.hasAvailability).to.equal(true);
```

Comment:
Can we add a date check in here to verify the dates are right if we find appointments?